sdk/core/core-http/test/policies/atomSerializationPolicyTests.ts

```diff
@@ -0,0 +1,162 @@
+    const request: WebResource = createRequest({
+      serializer: new TestSerializer()
+    });
+
+    const mockClient: HttpClient = {
+      sendRequest: (req) =>
+        Promise.resolve({
+          request: req,
+          status: 200,
+          headers: new HttpHeaders({
+            "content-type": "application/xml"
+          }),
+          bodyAsText: `<entry xmlns="http://www.w3.org/2005/Atom"><id>https://dummy.servicebus.windows.net/dummy?api-version=2017-04</id><title type="text">dummy</title><author><name>dummy</name></author><content type="application/xml"><QueueDescription xmlns="http://schemas.microsoft.com/netservices/2010/10/servicebus/connect" xmlns:i="http://www.w3.org/2001/XMLSchema-instance"><LockDuration>PT2M</LockDuration><MaxSizeInMegabytes>1024</MaxSizeInMegabytes></QueueDescription></content></entry>`
+        })
+    };
+
+    const policy = atomSerializationPolicy().create(mockClient, new RequestPolicyOptions());
+    await policy.sendRequest(request);
+  });
+});
+
+function createRequest(atomXmlOperationSpec?: AtomXmlOperationSpec): WebResource {
+  const request = new WebResource();
+  request.atomXmlOperationSpec = atomXmlOperationSpec;
+  return request;
+}
+
+class TestSerializer implements AtomXmlSerializer {
+  serialize(resource: any): object {
+    const property1 = "LockDuration";
+    const property2 = "MaxSizeInMegabytes";
+
+    const resourceName = "QueueDescription";
+    const content: any = {};
+    content[resourceName] = {
+      $: {
+        xmlns: "http://schemas.microsoft.com/netservices/2010/10/servicebus/connect"
+      }
+    };
+
+    content[resourceName][property1] = resource[property1];
+    content[resourceName][property2] = resource[property2];
+    content[Constants.XML_METADATA_MARKER] = { type: "application/xml" };
+
+    return {
+      entry: {
+        $: {
+          xmlns: "http://www.w3.org/2005/Atom"
+        },
+        updated: new Date().toISOString(),
+        content: content
+      }
+    };
+  }
+
+  async deserialize(response: HttpOperationResponse): Promise<HttpOperationResponse> {
+    const expectedParsedBody = {
+      entry: {
+        $: {
+          xmlns: "http://www.w3.org/2005/Atom"
+        },
+        author: { name: "dummy" },
+        id: "https://dummy.servicebus.windows.net/dummy?api-version=2017-04",
+        title: { $: { type: "text" }, _: "dummy" },
+        content: {
+          $: {
+            type: "application/xml"
+          },
+          QueueDescription: {
+            $: {
+              xmlns: "http://schemas.microsoft.com/netservices/2010/10/servicebus/connect",
+              "xmlns:i": "http://www.w3.org/2001/XMLSchema-instance"
+            },
+            LockDuration: "PT2M",
+            MaxSizeInMegabytes: "1024"
+          }
+        }
+      }
+    };
+    assert.deepEqual(response.parsedBody, expectedParsedBody);
+    return new Promise((resolve) => {
```

Comment:
Nit:
You can just return response instead of wrapping it in a promise and calling resolve. That's because you're in an async function.

Reply:
Updated